### PR TITLE
amp-bind: Support amp-iframe

### LINKED ISSUE #7837: Expose amp-iframe[src] to Bind
May need additional scrutiny to make sure this is okay, but no red flags so far

### PR DESCRIPTION
**Pull request by choumx**
Fixes #7837.

- Support binding to `src` attribute of `amp-iframe`.

/to @aghassemi @kmh287 PTAL

### CODE REVIEW

**Comment by choumx** [comment]
@aghassemi Feel free to redirect this PR review to whomever. 😃

File: extensions/amp-bind/0.1/test/test-bind-integration.js [hunk 1]
```diff
@@ -390,4 +390,19 @@ describe.configure().retryOnSaucelabs().run('amp-bind', function() {
...
+    it('should support binding to src', () => {
+      const button = fixture.doc.getElementById('iframeButton');
+      const ampIframe = fixture.doc.getElementById('iframe');
+      // Force layout in case element is not in viewport.
+      ampIframe.implementation_.layoutCallback();
+      const iframe = ampIframe.querySelector('iframe');
+      expect(iframe.src).to.not.contain('bound');
+      button.click();
+      return waitForBindApplication().then(() => {
+        expect(iframe.src).to.contain('bound');
```
Comment (kmh287) [inline extensions/amp-bind/0.1/test/test-bind-integration.js]: Also assert that the amp-iframe's src attribute has changed

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 2]
```diff
@@ -416,6 +416,20 @@ export class AmpIframe extends AMP.BaseElement {
...
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const src = mutations['src'];
+    if (src !== undefined) {
+      const iframeSrc = this.transformSrc_(src);
+      if (this.iframe_) {
+        this.iframe_.src = this.assertSource(
+            iframeSrc, window.location.href, this.sandbox_);
+      } else {
```
Comment (aghassemi) [inline extensions/amp-iframe/0.1/amp-iframe.js]: Let's set `this.iframeSrc = iframeSrc;` in both cases. No reason for it to fall behind the true src.

File: extensions/amp-iframe/0.1/test/test-amp-iframe.js [hunk 3]
```diff
@@ -645,4 +645,21 @@ describe('amp-iframe', () => {
...
+      height: 100,
+    }).then(amp => {
+      const container = amp.container;
+      const impl = container.implementation_;
+      const iframe = amp.iframe;
+
+      const newSrc = 'https://foo.bar';
+      container.setAttribute('src', newSrc);
+      impl.mutatedAttributesCallback({src: newSrc});
+      expect(iframe.getAttribute('src')).to.contain(newSrc);
```
Comment (aghassemi) [inline extensions/amp-iframe/0.1/test/test-amp-iframe.js]: after the change above, also assert that  `this.iframeSrc` is changed.

File: extensions/amp-bind/0.1/test/test-bind-integration.js (hunk 1 above)
Comment (choumx) [inline extensions/amp-bind/0.1/test/test-bind-integration.js]: Done.

File: extensions/amp-iframe/0.1/test/test-amp-iframe.js (hunk 3 above)
Comment (choumx) [inline extensions/amp-iframe/0.1/test/test-amp-iframe.js]: Done.

File: extensions/amp-iframe/0.1/amp-iframe.js (hunk 2 above)
Comment (choumx) [inline extensions/amp-iframe/0.1/amp-iframe.js]: Misread the code and thought it was set to null after load. Looks like `onload` is called again after changing the `src` anyways. Thanks!

**Pull request merged**